### DISCUSSION TITLE
Tailwind ring not working without ring-inset

### DISCUSSION THREAD

**Discussion by JamesHolt1995** [tailwindlabs/tailwindcss#12598] [Help] [answered]
`class="bg-cover rounded-full w-20 h-20 ring-offset-4 ring-2" `
The above shows nothing. If I add `ring-inset` it does work but with an inset shadow.

if I add the `--tw-ring-inset: ;` directly to `style=` of the element I'm styling it does work without insetting the shadow.

This happens on Chrome and Safari

**Comment by Zehad-10** [comment]
Hi

are you using `prefix` `tw-` in your code?  if yes then modify your classes with `tw` prefix. 

`ring-1 ring-inset ring-gray-300 focus:ring-2 focus:ring-inset focus:ring-indigo-600`

**Comment by JamesHolt1995** [comment, marked as answer]
This had to do with my Gulp setup. I was using Clean CSS and it was stripping all empty css variables from my css bundle. I've switched to CSS Nano to solve this issue
